addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -134,12 +139,27 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 sourceElements.stream()
                         .filter(element -> SupportedElements.contains(element.getName()))
                         .filter(isNotTrusted(trustedDomains, msg.getRequestHeader().getHostName()))
+                        .filter(
+                                ele ->
+                                        HTMLElementName.SCRIPT.equalsIgnoreCase(ele.getName())
+                                                || (HTMLElementName.LINK.equalsIgnoreCase(
+                                                                ele.getName())
+                                                        && linkRelNullOrRelevant(
+                                                                ele.getAttributeValue("rel"))))
```

Comment:
nit: consider extracting the full lambda into a method that includes the `nullOrRelevant` check, e.g.

```java
private static boolean isElementNotRelevant(Element element) {
    if (HTMLElementName.SCRIPT.equalsIgnoreCase(element.getName())) {
        return true;
    }
    if (!HTMLElementName.LINK.equalsIgnoreCase(element.getName())) {
        return false;
    }
    String relValue = element.getAttributeValue("rel");
    if (relValue == null) {
        return true;
    }
    return Arrays.stream(relValue.toLowerCase(Locale.ROOT).split(" "))
        .anyMatch(LINK_REL_KEYWORDS_FOR_SRI::contains);
}
```

and then the filter can just be

```java
.filter(SubResourceIntegrityAttributeScanRule::isElementNotRelevant)
```

Reply:
I'd suggest `SupportedElements.isRelevant(Element)` instead of `contains(Sring)` and have the attribute(s) as a property of each enum value (would avoid duplicating the name checks again).

Reply:
Thank you both, I like it 👍

Reply:
The tag should not be removed from the supported element.

Reply:
Ok I've restored that.

I trimmed it because it was and is unused.

Reply:
It should not be unused, it should still iterate all supported elements as before and check its tag and attributes.

Reply:
Sorry I'm not seeing what you're after

Reply:
Do you mean restoring the filter/contains on the stream? I thought you didn't want to check names/tags twice?

Reply:
Okay, I made it used with a single check. Seems very artificial but that's fine I guess.

Reply:
We should not be hardcoding checks unnecessarily, e.g.:
```java
public static boolean anyMatch(Element element) {
    return Stream.of(values()).anyMatch(e -> e.isRelevant(element));
}

private boolean isRelevant(Element element) {
    if (!tag.equalsIgnoreCase(element.getName())) {
        return false;
    }
    if (relevantAttributes.isEmpty()) {
        return true;
    }
    for (var entry : relevantAttributes.entrySet()) {
        String attributeValue = element.getAttributeValue(entry.getKey());
        if (attributeValue == null
                || Arrays.stream(attributeValue.toLowerCase(Locale.ROOT).split(" "))
                        .anyMatch(entry.getValue()::contains)) {
            return true;
        }
    }
    return false;
}
```